```{r include = FALSE}
require(rmarkdown); require(knitr); require(sevenbridges); require(readr)

options(max.print = "99"); opts_knit$set(width = 99)
opts_chunk$set(cache=FALSE, tidy=FALSE)

source("~/sbg_auth.R")

```

<hr>

Cancer Genome Project Insertion/Deletion detection pipeline based around Pindel

https://github.com/cancerit/cgpPindel

## base command

```{sh base command, eval=FALSE}

pindel.pl
-t [tumour mutant bam]
-n [normal wildtype bam]
-r [genome reference]
-c $CPU
-sp HUMAN 
-as GRCh37 
-st WGS
-e NC_007605,hs37d5,GL%
-s pindel_simpleRepeats.bed.gz
-f pindel_genomicRules.lst
-g pindel_human.GRCh37.indelCoding.bed.gz
-u pindel_pindel_np.gff3.gz
-sf pindel_softRules.lst
-b brass-ucscHiDepth_0.01_mrg1000_no_exon_coreChrs.bed.gz
-o ./pindel_output

```

<br><br><br><br>

## docker 

The docker conatiner is

``` cgrlab/cgpbox_dev:develop ```

## App ports {.tabset .tabset-fade .tabset-tabs}

### Inputs

```{r inputs, eval=run.mode=="push"}
inputs = list(
  

input(id = "tumour_bam", label = "tumour_bam", description = "tumour bam", type = "File", secondaryFiles = list(".bai"), prefix = "-t"),

input(id = "normal_bam", label = "normal_bam", description = "wildtype bam", type = "File", secondaryFiles = list(".bai"), prefix = "-n"),

input(id = "reference", label = "reference", description = "reference", type = "File", secondaryFiles = list(".fai"), prefix = "-r"),
  
input(id = "cgpbox_reference_files", label = "cgpbox_reference_files", description = "extracted reference files", type = "File...", stageInput = "link")

)

```

### Outputs

```{r output, eval = run.mode=="push"}

outputs = list(

output(id = "std_out", label = "std_out", 
description = "standard output", type = "File...",
metadata = list(org = "cgp"),
glob = Expression(engine = "#cwl-js-engine",
script = "\"*.{out}\"")),
  
output(id = "pindel_bed", label = "pindel_bed", 
description = "pindel_bed outputkl", type = "File",
inheritMetadataFrom = "#normal_bam", metadata = list(org = "cgp"),
glob = Expression(engine = "#cwl-js-engine",
script = "\"*.germline.bed\""))
  
)


```

## Define Tool Object and Push to Platform {.tabset .tabset-fade .tabset-tabs}
  
### Tool object

```{r 5 create tool object, eval=run.mode=="push"}

tool <- Tool(
id = "pindel", 
label = "pindel",
hints = requirements(docker(pull = "cgrlab/cgpbox_dev:develop"), cpu(1), mem(1000)),
baseCommand = "mkdir pindel_output && CPU=`grep -c ^processor /proc/cpuinfo` && pindel.pl -c $CPU -sp HUMAN -as GRCh37 -st WGS -e NC_007605,hs37d5,GL% -s pindel_simpleRepeats.bed.gz -f pindel_genomicRules.lst -g pindel_human.GRCh37.indelCoding.bed.gz -u pindel_pindel_np.gff3.gz -sf pindel_softRules.lst -b brass-ucscHiDepth_0.01_mrg1000_no_exon_coreChrs.bed.gz -o ./pindel_output",
stdout = "std.out",
inputs = inputs,
outputs = outputs)


```

### Push app to sbg platform

```{r 6 push app to sbg, eval=destination=="platform"}

p$app_add("pindel", tool)


```

