
```{r include = FALSE}
require(rmarkdown); require(knitr); require(sevenbridges); require(readr)

options(max.print = "99"); opts_knit$set(width = 99)
opts_chunk$set(cache=FALSE, tidy=FALSE)

source("~/sbg_auth.R")

```

<hr>

Somatic copy number analysis using paired end wholegenome sequencing

https://github.com/cancerit/ascatNgs

## base command

```{sh base command, eval=FALSE}

ascat.pl
-t [tumour bam]
-n [normal wildtype bam]
-r [genome reference]
-c [cpu number]
-rs 'HUMAN' 
-ra GRCh37 
-pr WGS
-pl ILLUMINA
-sg ascat_SnpGcCorrections.tsv
-q 20
-g L
-o ./ascat_output

```


## Prep ASCAT for caveman

```

&& perl -ne '@F=(split q{,}, $_)[1,2,3,4]; $F[1]-1; print join("\t",@F)."\n";' < copynumber.caveman.csv > ./normal_cn.bed && perl -ne '@F=(split q{,}, $_)[1,2,3,6]; $F[1]-1; print join("\t",@F)."\n";' < copynumber.caveman.csv > ./tumour_cn.bed

```


<br><br><br><br>

## docker 

The docker conatiner is

``` cgrlab/cgpbox_dev:develop ```


## App ports {.tabset .tabset-fade .tabset-tabs}

### Inputs

```{r inputs, eval=run.mode=="push"}

inputs = list(
  
input(id = "tumour_bam", label = "tumour_bam", description = "mutant tumour bam", type = "File", secondaryFiles = list(".bai"), prefix = "-t"),
  
input(id = "normal_bam", label = "normal_bam", description = "normal wildtype bam", type = "File", secondaryFiles = list(".bai"), prefix = "-n"),

input(id = "reference", label = "reference", description = "genome reference", type = "File", secondaryFiles = list(".fai"), prefix = "-r"),

input(id = "cgpbox_reference_files", label = "cgpbox_reference_files", description = "extracted reference files", type = "File...", stageInput = "link")

)

```

### Outputs

```{r output, eval = run.mode=="push"}

outputs = list(

output(id = "std_out", label = "std_out", 
description = "standard output", type = "File...",
metadata = list(org = "cgp"),
glob = Expression(engine = "#cwl-js-engine",
script = "\"*.{out}\"")),
  
output(id = "ascat_statistics", label = "ascat_statistics", 
description = "ascat_statistics", type = "File...",
inheritMetadataFrom = "#normal_bam", metadata = list(org = "cgp"),
glob = Expression(engine = "#cwl-js-engine",
script = "\"ascat_output/*.samplestatistics.txt\"")),
  
output(id = "tumour_copy_number_bed", label = "tumour_copy_number_bed", 
description = "tumour_copy_number_bed", type = "File...",
inheritMetadataFrom = "#tumour_bam", metadata = list(org = "cgp"),
glob = Expression(engine = "#cwl-js-engine",
script = "\"tumour_cn.bed\"")),
  
output(id = "normal_copy_number_bed", label = "normal_copy_number_bed", 
description = "normal_copy_number_bed", type = "File...",
inheritMetadataFrom = "#normal_bam", metadata = list(org = "cgp"),
glob = Expression(engine = "#cwl-js-engine",
script = "\"normal_cn.bed\""))

)


```

## Add on scripts {.tabset .tabset-fade .tabset-tabs}

```{r add on scripts, eval=run.mode=="push"}

prep_for_caveman_sh = fileDef(name = "prep_for_caveman.sh", content = read_file("prep_for_caveman.sh"))

```

## Define Tool Object and Push to Platform {.tabset .tabset-fade .tabset-tabs}
  
### Tool object

```{r 5 create tool object, eval=run.mode=="push"}

tool <- Tool(
id = "ASCAT", 
label = "ASCAT",
hints = requirements(docker(pull = "cgrlab/cgpbox_dev:develop"), cpu(1), mem(1000)),
requirements = requirements(prep_for_caveman_sh),

baseCommand = "CPU=`grep -c ^processor /proc/cpuinfo` && ascat.pl -c $CPU -rs Human -ra NCBI37 -pr WGS -pl ILLUMINA -sg ascat_SnpGcCorrections.tsv -q 20 -g L -o ascat_output",

stdout = "std.out",

inputs = inputs,
  
arguments = CCBList(CommandLineBinding(position = 99, valueFrom = list("\"&& sh prep_for_caveman.sh\""))), 
  
outputs = outputs)


```

### Push app to sbg platform

```{r 6 push app to sbg, eval=destination=="platform"}

p$app_add("ASCAT", tool)


```
